gramps/plugins/webreport/person.py

```diff
@@ -1635,6 +1638,55 @@ def display_ind_parent_family(self, birthmother, birthfather, family,
                 trow += tcell
                 table += trow
 
+    def step_or_not(self, family, handle):
+        """
+        Quickly check to see if this person is a stepmother or stepfather
+
+        We assume this is not a stepmother or a stepfather if :
+         1 - The father or mother died before the child birth
+         2 - The father or the mother divorced before the child birth
+         3 - The father or the mother married after the child death
+
+        In all other cases, they are stepfather or stepmother.
+
+        @param: family The family we examine
+        @param: handle The handle of the father or the mother
+        self.person    The child for whom we need this result
+        """
+        bd_date = Today()
+        bd_event = get_birth_or_fallback(self.r_db, self.person)
+        if bd_event:
+            bd_date = bd_event.get_date_object()
+        for event_ref in family.get_event_ref_list():
+            event = self.r_db.get_event_from_handle(event_ref.ref)
+            if event.type == EventType.DIVORCE and \
+                (event_ref.get_role() == EventRoleType.FAMILY or
```

Comment:
It would be more pythonic to write:

`event_ref.get_role() in (EventRoleType.FAMILY, EventRoleType.PRIMARY)`

Reply:
So the solution is : 
            if (event.type == EventType.MARRIAGE and
                (event_ref.get_role() in (EventRoleType.FAMILY, EventRoleType.PRIMARY ))):